src/main/scala/mesosphere/marathon/core/launcher/impl/InstanceOpFactoryImpl.scala

```diff
@@ -453,9 +454,13 @@ object InstanceOpFactoryImpl {
 
     taskIDs.map { taskId =>
       // the task level host ports are needed for fine-grained status/reporting later on
-      val taskHostPorts: Seq[Int] = taskId.containerName.map { ctName =>
-        allocPortsByCTName.withFilter { case (name, port) => name == ctName }.map(_._2)
-      }.getOrElse(Seq.empty[Int])
+      val taskHostPorts: Seq[Int] = taskId match {
+        case Task.ResidentTaskId(_, Some(containerName), _) =>
```

Comment:
Why don't we keep `containerName` and move this `match` into it?

Reply:
`LegacyId` does not have a `containerName` field and I think it should not but then again it would reduce refactored code. I'll add it then.

Reply:
✅